packages/grid/x-data-grid/src/components/panel/GridColumnsPanel.tsx

```diff
@@ -70,6 +70,20 @@ export function GridColumnsPanel(props: GridColumnsPanelProps) {
   const ownerState = { classes: rootProps.classes };
   const classes = useUtilityClasses(ownerState);
 
+  const sortedColumns = React.useMemo(() => {
+    if (rootProps.componentsProps?.columnsPanel.sort === 'asc') {
```

Comment:
`rootProps.componentsProps?.columnsPanel` is already passed to the column panel. 

https://github.com/mui/mui-x/blob/33cf86590cd69cdcd94dbaa831c1aeb2e70785c0/packages/grid/x-data-grid/src/hooks/features/columns/useGridColumns.tsx#L390:L391
```jsx
const ColumnsPanel = props.components.ColumnsPanel;
return <ColumnsPanel {...props.componentsProps?.columnsPanel} />;
```

So you can simply get `sort` from the `props`

with 
```jsx
const { sort, ...other } = props
```

and replace the `...props` by `...other` in the `GridPanelWrapper`

Reply:
You can also add the `sort` property to the type definition to `GridColumnsPanelProps`. For the filter panel it looks like that 

```ts
/**
 * Changes how the options in the columns selector should be ordered.
 * If not specified, the order is derived from the `columns` prop.
 */
columnsSort?: 'asc' | 'desc';
```

This should allow TS autocompletion, and warning is `asc` or `desc` is miss spell

Reply:
+1 for Alexandre's suggestion. You also need to replace `props` with `other` in

https://github.com/mui/mui-x/blob/412d8db2feb1a9e14ecc9cbe4270cf25749b4678/packages/grid/x-data-grid/src/components/panel/GridColumnsPanel.tsx#L129 to avoid passing `sort` to the DOM element.

![image](https://user-images.githubusercontent.com/42154031/186904719-7d37fdb5-75aa-4f52-8f2d-8d1cacdd4ba3.png)